components/automate-deployment/pkg/server/automate_config.go

```diff
@@ -280,6 +281,46 @@ func setConfigForRedirectLogs(req *api.PatchAutomateConfigRequest, existingCopy
 		}
 	}
 
+	//Set the config if already not set
+	if req.GetConfig().GetGlobal().GetV1().GetLog().GetRedirectSysLog().GetValue() == true &&
+		existingCopy.GetGlobal().GetV1().GetLog().GetRedirectSysLog().GetValue() == true {
+
+		if req.GetConfig().GetGlobal().GetV1().GetLog().GetRedirectLogFilePath().GetValue() == existingCopy.GetGlobal().GetV1().GetLog().GetRedirectLogFilePath().GetValue() {
+			res, err := UpdateOfLogroateConfig(req, existingCopy)
```

Comment:
if we are calling this function in both blocks? can't we call it once?

Reply:
We need it at both the places as we have two different conditions.
- if path is the same just update logtorate config
- if path is different then rsyslog and logrotate